src/azureFunction/azureFunctionUtils.ts

```diff
@@ -0,0 +1,215 @@
+}
+
+/**
+ * Gets the azure function project for the user to choose from a list of projects files
+ * If only one project is found that project is used to add the binding to
+ * if no project is found, user is informed there needs to be a C# Azure Functions project
+ * @returns the selected project file path
+ */
+export async function getAzureFunctionProject(): Promise<string | undefined> {
+	let selectedProjectFile: string | undefined = '';
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return selectedProjectFile;
+	} else {
+		const projectFiles = await getAzureFunctionProjectFiles();
+		if (projectFiles !== undefined) {
+			if (projectFiles.length > 1) {
+				// select project to add azure function to
+				selectedProjectFile = (await vscode.window.showQuickPick(projectFiles, {
+					canPickMany: false,
+					title: constants.selectProject,
+					ignoreFocusOut: true
+				}));
+				return selectedProjectFile;
+			} else if (projectFiles.length === 1) {
+				// only one azure function project found
+				return projectFiles[0];
+			}
+		}
+		return undefined;
+	}
+}
+
+/**
+ * Gets the azure function project files based on the host file found in the same folder
+ * @returns the azure function project files paths
+ */
+export async function getAzureFunctionProjectFiles(): Promise<string[] | undefined> {
+	let projFiles: string[] = [];
+	const hostFiles = await getHostFiles();
+	for (let host of hostFiles) {
+		let projectFile = await vscode.workspace.findFiles('*.csproj', path.dirname(host));
+		projectFile.filter(file => path.dirname(file.fsPath) === path.dirname(host) ? projFiles.push(file?.fsPath) : projFiles);
+	}
+	return projFiles.length > 0 ? projFiles : undefined;
+}
+
+/**
+ * Gets the host files from the workspace
+ * @returns the host file paths
+ */
+export async function getHostFiles(): Promise<string[] | undefined> {
+	const hostUris = await vscode.workspace.findFiles('**/host.json');
+	const hostFiles = hostUris.map(uri => uri.fsPath);
+	return hostFiles.length > 0 ? hostFiles : undefined;
+}
+
+/**
+ * Gets the local.settings.json file path
+ * @param projectFile path of the azure function project
+ * @returns the local.settings.json file path
+ */
+export async function getSettingsFile(projectFile: string): Promise<string | undefined> {
+	return path.join(path.dirname(projectFile), 'local.settings.json');
+}
+
+/**
+ * Retrieves the new function file once the file is created
+ * @param projectFile is the path to the project file
+ * @returns the function file path once created
+ */
+export function waitForNewFunctionFile(projectFile: string): Promise<string> {
+	return new Promise((resolve, reject) => {
+		const watcher = vscode.workspace.createFileSystemWatcher((
+			path.dirname(projectFile), '**/*.cs'), false, true, true);
+		const timeout = setTimeout(async () => {
+			reject('Timed out waiting for azure function file creation');
```

Comment:
This should actually be `reject(new Error('Timed out waiting for Azure Function file creation'));` (we should always throw errors, not strings directly)

And if the rejection is displayed to the user I would suggest localizing the string as well.